Libplanet.RocksDBStore/RocksDBStore.cs

```diff
@@ -422,16 +495,28 @@ public override IEnumerable<HashDigest<SHA256>> IterateBlockHashes()
             }
 
             byte[] key = BlockKey(blockHash);
-            byte[] bytes = _blockDb.Get(key);
+            byte[] epochBytes = _blockIndexDb.Get(key);
 
-            if (bytes is null)
+            if (epochBytes is null)
             {
                 return null;
             }
 
-            BlockDigest blockDigest = BlockDigest.Deserialize(bytes);
+            RocksDBStoreEpoch epoch = RocksDBStoreEpoch.Deserialize(epochBytes);
+
+            _rwBlockLock.EnterReadLock();
+            if (!_blockDbCache.TryGetValue(epoch.DbName, out RocksDb blockDb))
+            {
+                blockDb = RocksDBUtils.OpenRocksDb(_options, BlockDbPath(epoch.DbName));
+                _blockDbCache.AddOrUpdate(epoch.DbName, blockDb);
+            }
+
+            byte[] blockBytes = blockDb.Get(key);
+
+            BlockDigest blockDigest = BlockDigest.Deserialize(blockBytes);
 
             _blockCache.AddOrUpdate(blockHash, blockDigest);
+            _rwBlockLock.ExitReadLock();
```

Comment:
`_rwBlockLock.ExitReadLock()` should be inside a `finally` block so that it's guaranteed to be called even if an any error happens.

```suggestion
            try
            {
                if (!_blockDbCache.TryGetValue(epoch.DbName, out RocksDb blockDb))
                {
                    blockDb = RocksDBUtils.OpenRocksDb(_options, BlockDbPath(epoch.DbName));
                    _blockDbCache.AddOrUpdate(epoch.DbName, blockDb);
                }
    
                byte[] blockBytes = blockDb.Get(key);
    
                BlockDigest blockDigest = BlockDigest.Deserialize(blockBytes);
    
                _blockCache.AddOrUpdate(blockHash, blockDigest);
            }
            finally
            {
                _rwBlockLock.ExitReadLock();
            }
```